R/run_trial.R

```diff
@@ -149,6 +150,44 @@ run_trial <- function(
       )
     }
 
+    # post-processing to get evaluation metrics
+    if (!is.null(design$evaluation)){
+      for (design_type in names(design$evaluation)){
+        tmp_eval_design <- list(
+          type = design_type,
+          scheme = design$evaluation[[design_type]]
+        )
+        if (design_type %in% target_types_auc){
+          f_calc <- calc_auc_from_regimen
+        } else if (design_type %in% target_types_conc){
+          f_calc <- calc_concentration_from_regimen
+        } else {
+          warning(paste(design_type, "is not supported yet"))
```

Comment:
this is not a very user friendly method since you would generate a warning every patient.

if all methods accepted in `create_eval_design` are accepted in the code here, that's fine we don't have to handling that edge case. however, looking at `mipd_target_types()` it does not seem like you are supporting all methods yet.

perhaps instead the best place is to confirm that every single method in `create_eval_design` is supported throughout the code base, and instead put the warning *there* if the method is not supported since then the user would get the error only once and before they start a whole trial set up.